beacon-chain/blockchain/head.go

```diff
@@ -109,3 +110,96 @@ func (s *Service) saveHead(ctx context.Context, headRoot [32]byte) error {
 
 	return nil
 }
+
+// This gets called to update canonical root mapping. It does not save head block
+// root in DB. With the inception of inital-sync-cache-state flag, it uses finalized
+// check point as anchors to resume sync therefore head is no longer needed to be saved on per slot basis.
+func (s *Service) saveHeadNoDB(ctx context.Context, b *ethpb.SignedBeaconBlock, r [32]byte) error {
```

Comment:
this is an old function, I moved it from `service.go` to here, it makes more sense here